jubilant/_juju.py

```diff
@@ -127,6 +127,36 @@ def add_model(
         self.cli(*args, include_model=False)
         self.model = model
 
+    def add_secret(
+        self,
+        name: str,
+        content: Mapping[str, str],
```

Comment:
Wasn't there some way to create secrets with binary data?

Reply:
Yeah, you can do that with the `#base64` syntax (I'm not sure if that works inside a YAML file or not). However, we don't support that in Ops yet, so I don't think any charms are using it. I was going to leave it out for now.

Reply:
Perhaps the library users can supply binary secrets as encoded strings with `"#base64"` prefix even as it is now 🤔 

There's barely a couple of uses of python-libjuju `model.add_secret(...)` in the integration tests in our charm collection, and those are hardcoded strings.

And a single case of shelling out to set secrets, with `data: dict[str, str]`.

I guess you're right, it's very unlikely that someone needs binary secret data in integration tests.